fusor-ember-cli/app/controllers/rhev.js

```diff
@@ -27,35 +29,59 @@ export default Ember.Controller.extend(NeedsDeploymentMixin, {
   hasEngine: Ember.computed.alias('deploymentController.hasEngine'),
   hasNoEngine: Ember.computed.not('hasEngine'),
 
-  hasHypervisor: Ember.computed('deploymentController.model.discovered_hosts.[]', function() {
-    return (this.get('deploymentController.model.discovered_hosts.length') > 0);
-  }),
+  hasHypervisor: Ember.computed(
+    'deploymentController.model.discovered_hosts.[]',
+    function() {
+      return (this.get('deploymentController.model.discovered_hosts.length') > 0);
+    }
+  ),
   hasNoHypervisor: Ember.computed.not('hasHypervisor'),
 
   disableTabRhevHypervisors: Ember.computed(
     'deploymentController.model.rhev_is_self_hosted',
-    'hasNoEngine',
+    'validRhevEngine',
     function() {
-      return (!this.get('deploymentController.model.rhev_is_self_hosted') && this.get('hasNoEngine'));
+      return (!this.get('deploymentController.model.rhev_is_self_hosted') && !this.get('validRhevEngine'));
     }
   ),
 
   disableTabRhevConfiguration: Ember.computed(
     'deploymentController.model.rhev_is_self_hosted',
-    'hasNoEngine',
-    'hasNoHypervisor',
+    'validRhevEngine',
+    'validRhevHypervisor',
     function() {
-      return ((this.get('deploymentController.model.rhev_is_self_hosted') && this.get('hasNoEngine')) ||
-              (!this.get('deploymentController.model.rhev_is_self_hosted') && this.get('hasNoHypervisor'))
+      return ((this.get('deploymentController.model.rhev_is_self_hosted') && !this.get('validRhevEngine')) ||
+              (!this.get('deploymentController.model.rhev_is_self_hosted') && (!this.get('validRhevHypervisor') || !this.get('validRhevEngine')))
```

Comment:
@eriknelson Renaming setIfHostnameValid to setIfHostnameInvalid to properly match the functionality with the boolean and also replaced the self hosted lines with an alias to make it easier to read for the long line widths.